### ISSUE TITLE
Finetuning is so slow on bloomz-7b1

### ISSUE THREAD

**Issue by raihan0824** [tloen/alpaca-lora#145]
Hi,

Are there any configurations for other models than llama? When I first tried to run the finetune script, I had this error:
`ValueError: Target modules ['q_proj', 'v_proj'] not found in the base model. Please check the target modules and try again.`

Then I change the TARGET_MODULES to:
``` python
TARGET_MODULES = [
    "query_key_value"
]
```

The script  run, but it's so slow
![Screenshot 2023-03-24 at 04 10 20](https://user-images.githubusercontent.com/84271135/227362368-28c7cda3-0b3a-4fe0-bcce-ab91666c55d8.png)
Tooks about 15 hours to finish.

I use A100 40GB, it should be faster than yours. Am I missing something here?

**Comment by raihan0824** [comment]
When I used the original script (using `decapoda-research/llama-7b-hf` model), it still so slow compared to yours.

![Screenshot 2023-03-24 at 22 32 41](https://user-images.githubusercontent.com/84271135/227570731-cc03f063-d330-4390-acdc-c94132957248.png)

Tooks about 9 hours to finish

**Comment by raihan0824** [comment]
any help on this?

**Comment by scarydemon2** [comment]
hi, do you met error "expected scalar type Half but found Float" after change model to bloomz?

**Comment by raihan0824** [comment]
no, I only get this error: `ValueError: Target modules ['q_proj', 'v_proj'] not found in the base model. Please check the target modules and try again.` Did you able to do something different?

**Comment by satani99** [comment]
@raihan0824 how did you get the target modules for any particular model?

**Comment by satani99** [comment]
from transformers import GPTNeoXForCausalLM

model = GPTNeoXForCausalLM.from_pretrained("EleutherAI/gpt-neox-20b")

for name, param in model.named_parameters():
print(name, param.shape)


For any model
